In [104]:
import pandas as pd
import sqlite3
conn = sqlite3.connect('ax.db')
query = 'SELECT * FROM dataDescribeSales'
df = pd.read_sql_query(query, conn)
conn.close()


In [105]:
df

Area Code          State          Market   Market Size  \
0  Store's Code  Store's State  Store's Region  Store's Size   

                    Profit                                             Margin  \
0  Profits in Dollars (\$)  Profit + Total Expenses (\$) OR Sales - COGS (\$)   

                           Sales                    COGS  \
0  Values Acquired in Sales (\$)  Cost of Goods Sold ($)   

                                   Total Expenses                   Marketing  \
0  Total Expenses to get the Product to Sell (\$)  Expenses in Marketing (\$)   

                                           Inventory         Budget Profit  \
0  Inventory Value of the Product in the Sale Mom...  Expected Profit (\$)   

          Budget COGS                                      Budget Margin  \
0  Expected COGS (\$)  Expected Profit + Expected Total Expenses (\$)...   

                            Budget Sales   ProductID       Date  \
0  Expected Value Acquired in Sales (\$)  Product ID  Sale Date   

       Product Type              Product  Type  
0  Product Category  Product Description  Type

In [106]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1 entries, 0 to 0
Data columns (total 20 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Area Code       1 non-null      object
 1   State           1 non-null      object
 2   Market          1 non-null      object
 3   Market Size     1 non-null      object
 4   Profit          1 non-null      object
 5   Margin          1 non-null      object
 6   Sales           1 non-null      object
 7   COGS            1 non-null      object
 8   Total Expenses  1 non-null      object
 9   Marketing       1 non-null      object
 10  Inventory       1 non-null      object
 11  Budget Profit   1 non-null      object
 12  Budget COGS     1 non-null      object
 13  Budget Margin   1 non-null      object
 14  Budget Sales    1 non-null      object
 15  ProductID       1 non-null      object
 16  Date            1 non-null      object
 17  Product Type    1 non-null      object
 18  Product       

- bảng này mô tả bán hàng, gồm có 20 cột

In [107]:
conn = sqlite3.connect('ax.db')
query1 = 'SELECT * FROM dataDescribeSuperMarketSales'
df1 = pd.read_sql_query(query1, conn)
conn.close()



In [108]:
df1

Invoice id  \
0  Computer generated sales slip invoice identifi...   

                                              Branch  \
0  Branch of supercenter (3 branches are availabl...   

                       City  \
0  Location of supercenters   

                                       Customer type                   Gender  \
0  Type of customers, recorded by Members for cus...  Gender type of customer   

                                        Product line  \
0  General item categorization groups - Electroni...   

                   Unit price                                  Quantity  \
0  Price of each product in $  Number of products purchased by customer   

                              Tax                      Total  \
0  5% tax fee for customer buying  Total price including tax   

                                                Date  \
0  Date of purchase (Record available from Januar...   

                          Time  \
0  Purchase time (10am to 9pm)   

                                             Payment                COGS  \
0  Payment used by customer for purchase (3 metho...  Cost of goods sold   

   Gross margin percentage  Gross income  \
0  Gross margin percentage  Gross income   

                                              Rating  
0  Customer stratification rating on their overal...

In [109]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1 entries, 0 to 0
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Invoice id               1 non-null      object
 1   Branch                   1 non-null      object
 2   City                     1 non-null      object
 3   Customer type            1 non-null      object
 4   Gender                   1 non-null      object
 5   Product line             1 non-null      object
 6   Unit price               1 non-null      object
 7   Quantity                 1 non-null      object
 8   Tax                      1 non-null      object
 9   Total                    1 non-null      object
 10  Date                     1 non-null      object
 11  Time                     1 non-null      object
 12  Payment                  1 non-null      object
 13  COGS                     1 non-null      object
 14  Gross margin percentage  1 non-null      objec

- bảng này mô tả bán hàng siêu thị gồm có 17 cột
- kiểu dữ liệu kiểu object

In [110]:
conn = sqlite3.connect('ax.db')
query2 = 'SELECT * FROM dataSales'
df2 = pd.read_sql_query(query2, conn)
conn.close()

In [111]:
df2

Area Code        State Market   Market Size  Profit  Margin  Sales  \
0           203  Connecticut   East  Small Market   107.0   176.0  292.0   
1           203  Connecticut   East  Small Market    75.0   135.0  225.0   
2           203  Connecticut   East  Small Market   122.0   195.0  325.0   
3           203  Connecticut   East  Small Market   105.0   174.0  289.0   
4           203  Connecticut   East  Small Market   104.0   135.0  223.0   
...         ...          ...    ...           ...     ...     ...    ...   
4243        985    Louisiana  South  Small Market    -6.0    66.0  113.0   
4244        985    Louisiana  South  Small Market    -7.0    61.0  105.0   
4245        985    Louisiana  South  Small Market    -3.0    76.0  131.0   
4246        985    Louisiana  South  Small Market     1.0    87.0  160.0   
4247        985    Louisiana  South  Small Market     4.0    90.0  153.0   

       COGS  Total Expenses  Marketing  Inventory  Budget Profit  Budget COGS  \
0     116.0            69.0       38.0      962.0          110.0        110.0   
1      90.0            60.0       29.0     1148.0           90.0         80.0   
2     130.0            73.0       42.0     1134.0          130.0        110.0   
3     115.0            69.0       37.0     1166.0          110.0        100.0   
4      90.0            56.0       29.0     1148.0           90.0         80.0   
...     ...             ...        ...        ...            ...          ...   
4243   47.0            72.0       42.0     -395.0            0.0         40.0   
4244   44.0            68.0       39.0     -508.0           10.0         30.0   
4245   55.0            79.0       49.0    -1053.0            0.0         40.0   
4246   63.0            86.0       57.0       37.0            0.0         50.0   
4247   64.0            81.0       58.0     -113.0           10.0         50.0   

      Budget Margin  Budget Sales  ProductId               Date Product Type  \
0             160.0         270.0          2  04/01/10 00:00:00       Coffee   
1             130.0         210.0          2  07/01/10 00:00:00       Coffee   
2             180.0         290.0          2  11/01/10 00:00:00       Coffee   
3             160.0         260.0          2  12/01/10 00:00:00       Coffee   
4             130.0         210.0          2  07/01/11 00:00:00       Coffee   
...             ...           ...        ...                ...          ...   
4243           50.0          90.0          4  07/01/10 00:00:00     Espresso   
4244           60.0          90.0          4  08/01/10 00:00:00     Espresso   
4245           60.0         100.0          4  12/01/10 00:00:00     Espresso   
4246           70.0         120.0          4  04/01/11 00:00:00     Espresso   
4247           80.0         130.0          4  05/01/11 00:00:00     Espresso   

          Product     Type  
0       Columbian  Regular  
1       Columbian  Regular  
2       Columbian  Regular  
3       Columbian  Regular  
4       Columbian  Regular  
...           ...      ...  
4243  Caffe Latte  Regular  
4244  Caffe Latte  Regular  
4245  Caffe Latte  Regular  
4246  Caffe Latte  Regular  
4247  Caffe Latte  Regular  

[4248 rows x 20 columns]

In [112]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4248 entries, 0 to 4247
Data columns (total 20 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Area Code       4248 non-null   int64  
 1   State           4248 non-null   object 
 2   Market          4248 non-null   object 
 3   Market Size     4248 non-null   object 
 4   Profit          4248 non-null   float64
 5   Margin          4248 non-null   float64
 6   Sales           4248 non-null   float64
 7   COGS            4248 non-null   float64
 8   Total Expenses  4248 non-null   float64
 9   Marketing       4248 non-null   float64
 10  Inventory       4248 non-null   float64
 11  Budget Profit   4248 non-null   float64
 12  Budget COGS     4248 non-null   float64
 13  Budget Margin   4248 non-null   float64
 14  Budget Sales    4248 non-null   float64
 15  ProductId       4248 non-null   int64  
 16  Date            4248 non-null   object 
 17  Product Type    4248 non-null   o

- bảng này mô tả dữ liệu bán hàng gồm 4248 dòng và 20 cột
- có 3 kiểu dữ liệu : float64, int64, object
- các thuộc tính trong bảng dữ liệu không bị thiếu

In [113]:
df2.isnull().sum() #đếm giá trị thiếu trong từng cột -> không có giá trị thiếu 

Area Code         0
State             0
Market            0
Market Size       0
Profit            0
Margin            0
Sales             0
COGS              0
Total Expenses    0
Marketing         0
Inventory         0
Budget Profit     0
Budget COGS       0
Budget Margin     0
Budget Sales      0
ProductId         0
Date              0
Product Type      0
Product           0
Type              0
dtype: int64

In [114]:
df2.duplicated().sum() # không có giá trị trùng lặp

0

In [115]:
# kiểm tra giá trị cột 'Profit' =  sale (tổng doanh số) - COGS (giá vốn bán) - Total Expense (tổng chi phí)
for i in range(len(df2)):
   if( df2['Sales'][i]- df2['COGS'][i] - df2['Total Expenses'][i] != df2["Profit"][i]):
      print("test")

test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test


In [116]:
# cập nhật lại Profit ở bảng dữ liệu df2
df2['Profit'] = df2['Sales'] - df2['COGS'] - df2['Total Expenses']

In [117]:
# kiểm tra cột 'Margin' =  cột 'Profit' / cột 'Sales' không
for i in range(len(df2)):
   if( df2["Profit"][i] / df2["Sales"][i]!=df2["Margin"][i]):
      print("test")

test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test


In [118]:
# cập nhật lại cột Margin 
df2["Margin"] = df2["Profit"] / df2["Sales"] 

In [119]:
# kiểm tra xem tổng lợi nhuận (Profit) và tổng chi phí (Total Expenses) = giá trị biên ngân sách (Budget Margin) hay không
for i in range(len(df2)):
   if( df2["Profit"][i]+df2["Total Expenses"][i]!=df2["Budget Margin"][i]):
      print("test")

test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test


In [120]:
# cập nhật lại cột Budget Margin
df2["Budget Margin"] = df2["Profit"] + df2["Total Expenses"]

In [121]:
# Chuyển đổi cột 'Date' thành định dạng ngày tháng hợp lệ
df2['Date'] = pd.to_datetime(df2['Date'], errors='coerce')

In [123]:
df2.head(5)

Area Code        State Market   Market Size  Profit    Margin  Sales  \
0        203  Connecticut   East  Small Market   107.0  0.366438  292.0   
1        203  Connecticut   East  Small Market    75.0  0.333333  225.0   
2        203  Connecticut   East  Small Market   122.0  0.375385  325.0   
3        203  Connecticut   East  Small Market   105.0  0.363322  289.0   
4        203  Connecticut   East  Small Market    77.0  0.345291  223.0   

    COGS  Total Expenses  Marketing  Inventory  Budget Profit  Budget COGS  \
0  116.0            69.0       38.0      962.0          110.0        110.0   
1   90.0            60.0       29.0     1148.0           90.0         80.0   
2  130.0            73.0       42.0     1134.0          130.0        110.0   
3  115.0            69.0       37.0     1166.0          110.0        100.0   
4   90.0            56.0       29.0     1148.0           90.0         80.0   

   Budget Margin  Budget Sales  ProductId       Date Product Type    Product  \
0          176.0         270.0          2 2010-04-01       Coffee  Columbian   
1          135.0         210.0          2 2010-07-01       Coffee  Columbian   
2          195.0         290.0          2 2010-11-01       Coffee  Columbian   
3          174.0         260.0          2 2010-12-01       Coffee  Columbian   
4          133.0         210.0          2 2011-07-01       Coffee  Columbian   

      Type  
0  Regular  
1  Regular  
2  Regular  
3  Regular  
4  Regular

In [124]:
df2.describe()

Area Code       Profit       Margin        Sales         COGS  \
count  4248.000000  4248.000000  4248.000000  4248.000000  4248.000000   
mean    582.278013    54.490819     0.145015   192.987524    84.433145   
std     221.140310    83.970778     1.251241   151.133127    67.249769   
min     203.000000  -430.000000   -22.631579    17.000000     0.000000   
25%     417.000000    16.000000     0.162791   100.000000    43.000000   
50%     573.000000    35.000000     0.301887   138.000000    60.000000   
75%     772.000000    83.000000     0.373626   230.000000   100.000000   
max     985.000000   564.000000     0.851290   912.000000   364.000000   

       Total Expenses    Marketing    Inventory  Budget Profit  Budget COGS  \
count     4248.000000  4248.000000  4248.000000    4248.000000  4248.000000   
mean        54.063559    31.185028   749.381356      60.913371    74.830508   
std         32.352598    27.023264   661.031896      79.546123    66.238145   
min         10.000000     0.000000 -3534.000000    -320.000000     0.000000   
25%         33.000000    13.000000   432.000000      20.000000    30.000000   
50%         46.000000    22.000000   619.000000      40.000000    50.000000   
75%         65.000000    39.000000   910.500000      80.000000    90.000000   
max        190.000000   156.000000  8252.000000     560.000000   450.000000   

       Budget Margin  Budget Sales    ProductId  
count    4248.000000   4248.000000  4248.000000  
mean      108.554379    175.649718     6.887006  
std        96.646250    148.891522     3.664072  
min      -283.000000      0.000000     1.000000  
25%        55.000000     80.000000     4.000000  
50%        80.000000    130.000000     6.000000  
75%       139.000000    210.000000    10.000000  
max       613.000000   1140.000000    13.000000

In [125]:
df2['State'].unique()

array(['Connecticut', 'Washington', 'California', 'Texas', 'New York',
       'Ohio', 'Illinois', 'Louisiana', 'Florida', 'Wisconsin',
       'Colorado', 'Missouri', 'Iowa', 'Massachusetts', 'Oklahoma',
       'Utah', 'Oregon', 'New Mexico', 'New Hampshire', 'Nevada'],
      dtype=object)

In [52]:
conn = sqlite3.connect('ax.db')
query3 = 'SELECT * FROM data_supermarket_sales'
df3 = pd.read_sql_query(query2, conn)
conn.close()

In [53]:
df3.head(5)

Area Code        State Market   Market Size  Profit  Margin  Sales   COGS  \
0        203  Connecticut   East  Small Market   107.0   176.0  292.0  116.0   
1        203  Connecticut   East  Small Market    75.0   135.0  225.0   90.0   
2        203  Connecticut   East  Small Market   122.0   195.0  325.0  130.0   
3        203  Connecticut   East  Small Market   105.0   174.0  289.0  115.0   
4        203  Connecticut   East  Small Market   104.0   135.0  223.0   90.0   

   Total Expenses  Marketing  Inventory  Budget Profit  Budget COGS  \
0            69.0       38.0      962.0          110.0        110.0   
1            60.0       29.0     1148.0           90.0         80.0   
2            73.0       42.0     1134.0          130.0        110.0   
3            69.0       37.0     1166.0          110.0        100.0   
4            56.0       29.0     1148.0           90.0         80.0   

   Budget Margin  Budget Sales  ProductId               Date Product Type  \
0          160.0         270.0          2  04/01/10 00:00:00       Coffee   
1          130.0         210.0          2  07/01/10 00:00:00       Coffee   
2          180.0         290.0          2  11/01/10 00:00:00       Coffee   
3          160.0         260.0          2  12/01/10 00:00:00       Coffee   
4          130.0         210.0          2  07/01/11 00:00:00       Coffee   

     Product     Type  
0  Columbian  Regular  
1  Columbian  Regular  
2  Columbian  Regular  
3  Columbian  Regular  
4  Columbian  Regular

In [54]:
df3.info() 


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4248 entries, 0 to 4247
Data columns (total 20 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Area Code       4248 non-null   int64  
 1   State           4248 non-null   object 
 2   Market          4248 non-null   object 
 3   Market Size     4248 non-null   object 
 4   Profit          4248 non-null   float64
 5   Margin          4248 non-null   float64
 6   Sales           4248 non-null   float64
 7   COGS            4248 non-null   float64
 8   Total Expenses  4248 non-null   float64
 9   Marketing       4248 non-null   float64
 10  Inventory       4248 non-null   float64
 11  Budget Profit   4248 non-null   float64
 12  Budget COGS     4248 non-null   float64
 13  Budget Margin   4248 non-null   float64
 14  Budget Sales    4248 non-null   float64
 15  ProductId       4248 non-null   int64  
 16  Date            4248 non-null   object 
 17  Product Type    4248 non-null   o

In [56]:
df3.duplicated().sum() # không có giá trị trùng lặp

0

In [71]:
# kiểm tra giá trị cột 'Profit' =  sale (tổng doanh số) - COGS (giá vốn bán) - Total Expense (tổng chi phí)
for i in range(len(df3)):
   if( df3['Sales'][i]- df3['COGS'][i] - df3['Total Expenses'][i] != df3["Profit"][i]):
      print("test")

test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test


- giá trị bảng profit bị tính sai

In [72]:
#cập nhật lại profit
df3['Profit'] = df3['Sales'] - df3['COGS'] - df3['Total Expenses']

In [ ]:
# kiểm tra cột 'Margin' =  cột 'Profit' / cột 'Sales' không
for i in range(len(df3)):
   if( df3["Profit"][i] / df3["Sales"][i]!=df3["Margin"][i]):
      print("test")

In [ ]:
# cập nhật lại cột Margin
df['Margin'] = df['Profit'] / df['Sales']

In [62]:
# kiểm tra xem tổng lợi nhuận (Profit) và tổng chi phí (Total Expenses) = giá trị biên ngân sách (Budget Margin) hay không
for i in range(len(df3)):
   if( df3["Profit"][i]+df3["Total Expenses"][i]!=df3["Budget Margin"][i]):
      print("test")

test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test
test


- giá trị Budget Margin tính sai

In [68]:
# cập nhật lại cột Budget Margin
df3["Budget Margin"] = df3["Profit"] + df3["Total Expenses"]

In [76]:
# Tìm giá trị bị thiếu trong bảng df3
df3.isnull().sum()

Area Code         0
State             0
Market            0
Market Size       0
Profit            0
Margin            0
Sales             0
COGS              0
Total Expenses    0
Marketing         0
Inventory         0
Budget Profit     0
Budget COGS       0
Budget Margin     0
Budget Sales      0
ProductId         0
Date              0
Product Type      0
Product           0
Type              0
dtype: int64

In [78]:
# Chuyển đổi cột 'Date' thành định dạng ngày tháng hợp lệ
df3['Date'] = pd.to_datetime(df3['Date'], errors='coerce')


In [81]:
df3

Area Code        State Market   Market Size  Profit  Margin  Sales  \
0           203  Connecticut   East  Small Market   107.0   176.0  292.0   
1           203  Connecticut   East  Small Market    75.0   135.0  225.0   
2           203  Connecticut   East  Small Market   122.0   195.0  325.0   
3           203  Connecticut   East  Small Market   105.0   174.0  289.0   
4           203  Connecticut   East  Small Market    77.0   160.0  223.0   
...         ...          ...    ...           ...     ...     ...    ...   
4243        985    Louisiana  South  Small Market    -6.0    66.0  113.0   
4244        985    Louisiana  South  Small Market    -7.0    61.0  105.0   
4245        985    Louisiana  South  Small Market    -3.0    76.0  131.0   
4246        985    Louisiana  South  Small Market    11.0    87.0  160.0   
4247        985    Louisiana  South  Small Market     8.0    85.0  153.0   

       COGS  Total Expenses  Marketing  Inventory  Budget Profit  Budget COGS  \
0     116.0            69.0       38.0      962.0          110.0        110.0   
1      90.0            60.0       29.0     1148.0           90.0         80.0   
2     130.0            73.0       42.0     1134.0          130.0        110.0   
3     115.0            69.0       37.0     1166.0          110.0        100.0   
4      90.0            56.0       29.0     1148.0           90.0         80.0   
...     ...             ...        ...        ...            ...          ...   
4243   47.0            72.0       42.0     -395.0            0.0         40.0   
4244   44.0            68.0       39.0     -508.0           10.0         30.0   
4245   55.0            79.0       49.0    -1053.0            0.0         40.0   
4246   63.0            86.0       57.0       37.0            0.0         50.0   
4247   64.0            81.0       58.0     -113.0           10.0         50.0   

      Budget Margin  Budget Sales  ProductId       Date Product Type  \
0             176.0         270.0          2 2010-04-01       Coffee   
1             135.0         210.0          2 2010-07-01       Coffee   
2             195.0         290.0          2 2010-11-01       Coffee   
3             174.0         260.0          2 2010-12-01       Coffee   
4             160.0         210.0          2 2011-07-01       Coffee   
...             ...           ...        ...        ...          ...   
4243           66.0          90.0          4 2010-07-01     Espresso   
4244           61.0          90.0          4 2010-08-01     Espresso   
4245           76.0         100.0          4 2010-12-01     Espresso   
4246           87.0         120.0          4 2011-04-01     Espresso   
4247           85.0         130.0          4 2011-05-01     Espresso   

          Product     Type  
0       Columbian  Regular  
1       Columbian  Regular  
2       Columbian  Regular  
3       Columbian  Regular  
4       Columbian  Regular  
...           ...      ...  
4243  Caffe Latte  Regular  
4244  Caffe Latte  Regular  
4245  Caffe Latte  Regular  
4246  Caffe Latte  Regular  
4247  Caffe Latte  Regular  

[4248 rows x 20 columns]

In [82]:
df3.describe()

Area Code       Profit       Margin        Sales         COGS  \
count  4248.000000  4248.000000  4248.000000  4248.000000  4248.000000   
mean    582.278013    54.490819   115.161252   192.987524    84.433145   
std     221.140310    83.970778   112.726998   151.133127    67.249769   
min     203.000000  -430.000000  -491.000000    17.000000     0.000000   
25%     417.000000    16.000000    57.000000   100.000000    43.000000   
50%     573.000000    35.000000    83.000000   138.000000    60.000000   
75%     772.000000    83.000000   149.000000   230.000000   100.000000   
max     985.000000   564.000000   824.000000   912.000000   364.000000   

       Total Expenses    Marketing    Inventory  Budget Profit  Budget COGS  \
count     4248.000000  4248.000000  4248.000000    4248.000000  4248.000000   
mean        54.063559    31.185028   749.381356      60.913371    74.830508   
std         32.352598    27.023264   661.031896      79.546123    66.238145   
min         10.000000     0.000000 -3534.000000    -320.000000     0.000000   
25%         33.000000    13.000000   432.000000      20.000000    30.000000   
50%         46.000000    22.000000   619.000000      40.000000    50.000000   
75%         65.000000    39.000000   910.500000      80.000000    90.000000   
max        190.000000   156.000000  8252.000000     560.000000   450.000000   

       Budget Margin  Budget Sales    ProductId  
count    4248.000000   4248.000000  4248.000000  
mean      115.161252    175.649718     6.887006  
std       112.726998    148.891522     3.664072  
min      -491.000000      0.000000     1.000000  
25%        57.000000     80.000000     4.000000  
50%        83.000000    130.000000     6.000000  
75%       149.000000    210.000000    10.000000  
max       824.000000   1140.000000    13.000000

In [84]:
df3['State'].unique()

array(['Connecticut', 'Washington', 'California', 'Texas', 'New York',
       'Ohio', 'Illinois', 'Louisiana', 'Florida', 'Wisconsin',
       'Colorado', 'Missouri', 'Iowa', 'Massachusetts', 'Oklahoma',
       'Utah', 'Oregon', 'New Mexico', 'New Hampshire', 'Nevada'],
      dtype=object)